lib/liquid/template.rb

```diff
@@ -144,6 +144,13 @@ def render!(*args)
     def tokenize(source)
       source = source.source if source.respond_to?(:source)
       return [] if source.to_s.empty?
+
+      if source =~ WhiteSpaceTrimCheck
+        WhiteSpaceTrimTags.each do |(regexp, string)|
+          source = source.gsub(regexp, string)
```

Comment:
Hmm. I'm a bit concerned that this is essentially doing 4 passes (once for each regexp) over the entire source. I can't imagine that to not be a considerable performance hit.